docs/experimental/examples/oauth2/globus_app.rst

```diff
@@ -1,5 +1,202 @@
+
+
+Once instantiated, an app can be passed to any service client using the init
+``app`` kwarg (e.g. ``TransferClient(app=my_app)``). Doing this will bind the app to the
+client, registering a default set of scopes requirements for the service client's
+resource server and configuring the app as the service client's auth provider.
+
+
+..  tab-set::
+
+    ..  tab-item:: UserApp
+
+        Construct a UserApp then bind it to a transfer client and a flows client.
+
+        ..  Note::
+
+            ``UserApp.__init__(...)`` also accepts a `client_secret` kwarg which must be
+            supplied for confidential clients.
+
+        ..  code-block:: python
+
+            import globus_sdk
+            from globus_sdk.experimental.globus_app import UserApp
+
+            CLIENT_ID = "..."
+            my_app = UserApp("my-user-app", client_id=CLIENT_ID)
+
+            transfer_client = globus_sdk.TransferClient(app=my_app)
+            flows_client = globus_Sdk.FlowsClient(app=my_app)
+
+
+    .. tab-item:: ClientApp
+
+        Construct a ClientApp then bind it to a transfer client and flows_client.
+
+        ..  Note::
+
+            ``ClientApp.__init__(...)`` does not allow omission of the `client_secret`
+            kwarg as native clients are not allowed.
+
+        ..  code-block:: python
+
+            import globus_sdk
+            from globus_sdk.experimental.globus_app import ClientApp
+
+            CLIENT_ID = "..."
+            CLIENT_SECRET = "..."
+            my_app = ClientApp("my-client-app", client_id=CLIENT_ID, client_secret=CLIENT_SECRET)
+
+            transfer_client = globus_sdk.TransferClient(app=my_app)
+            flows_client = globus_sdk.FlowsClient(app=my_app)
+
+
+Usage
+-----
+
+From this point, the app manages scope validation, token caching, and authorizer
+supply for any the clients it is bound to.
+
+In the above example, listing a client's or user's flows becomes as simple as:
+
+..  code-block:: python
+
+    flows = flows_client.list_flows()["flows"]
+
+If cached tokens are missing, expired, or otherwise insufficient (e.g. the first time
+you run the script), the app will automatically initiate an auth flow to acquire new
+tokens. For a UserApp, this will print out a URL to terminal with a prompt instructing a
+the user to follow the link and enter the code they're given back into the terminal. For
+a ClientApp, the app will retrieve tokens programmatically through an Auth API.
+
+Once this auth flow has finished, the app will cache tokens for future use and
+invocation of your requested method will proceed as expected.
+
+
+Manually Running Login Flows
+----------------------------
+
+While your app will automatically initiate and oversee auth flows as detected, sometimes
+the programmer knows timing better. To manually trigger a login flow, call
+``GlobusApp.run_login_flow(...)``. This will initiate a flow requesting new tokens
```

Comment:
Above, the phrase "auth flow" is used; I recommend using that phrasing consistently.

```suggestion
``GlobusApp.run_login_flow(...)``. This will initiate an auth flow requesting new tokens
```